docs/enhancements/feature-enablement.md

```diff
@@ -0,0 +1,78 @@
+---
+title: feature-enablement
+authors:
+  - "@nmagnezi"
+creation-date: 2022-05-02
+last-updated: 2022-05-02
+---
+
+# Feature Enablement
+
+The current OCM-based authorization scheme used by assisted service as a part of the SaaS operation does not support a rollout of certain features to a select number of users.
+We wish to allow the following:
+* An ability to expose selected group(s) of users to assisted-service functionality (in OCM, the term is `capability`).
+* The user grouping should be based on the organization ID.
+
+
+## Motivation
+
+Some features might be of interest to partners or early adopters.
+Exposing such features to a pre-determined group of users can help us to gather feedback and improve while keeping the other users with a production-level user experience.
+
+
+### Goals
+* Add a generic method to assisted service authorization middleware to assert against an organization's capability. The method should query OCM based on the payload and asserted capability name.
+* Implement a functionality-specific assertion for `ARM`.
+
+
+## Proposal
+
+
+Initially, the authorization middleware should have a method to check if the user belongs to an organization with a particular capability.
+Adding such a method would be the first step, both to assert organization capabilities, such as ARM support, and any other capability of interest in the future.
+
+The API endpoints that need ARM support feature enablement:
+
+### V2ListSupportedOpenshiftVersions API
+
+* [ReleaseImage](../../models/release_image.go) has a `CPUArchitecture` attribute.
+* Expected to filter out ARM version if that capability is not included for the user organization.
```

Comment:
iiuc, you're using V2ListSupportedOpenshiftVersions anyhow, and you just want to add the capability check in addition to it. That makes sense.

However, consider that it might make sense for V2ListSupportedOpenshiftVersions to return ARM if the openshift version supports it, and add another check for the capability.
Both would work. It is just a matter of how you want your API to behave like.

Reply:
Worth a discussion.
We need to think if it makes sense to have 2 APIs and what would be the use-case of each API in that case.
From what I understand, we aim to "hide" this functionality in case it is not allowed (thus the HTTP 400 response that Avishay mentioned above). Let's follow up on this.

Reply:
@oourfali 
Based on the other comments and the discussion we had: I understand that the expected behavior is to "hide" the CPU architectures that the user has no access rights to. Let me know if you think the same.